toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -314,105 +345,57 @@ func replaceKernelCommandLineArgValue(inputGrubCfgContent string, name string, v
 	return outputGrubCfgContent, oldValue, nil
 }
 
-// Inserts new kernel command-line args into the grub config file.
-func addKernelCommandLine(kernelExtraArgs imagecustomizerapi.KernelExtraArguments,
-	imageChroot *safechroot.Chroot,
-) error {
-	var err error
-
-	extraCommandLine := strings.TrimSpace(string(kernelExtraArgs))
-	if extraCommandLine == "" {
-		// Nothing to do.
-		return nil
-	}
-
-	logger.Log.Infof("Setting KernelCommandLine.ExtraCommandLine")
-
-	grub2ConfigFile, err := readGrub2ConfigFile(imageChroot)
-	if err != nil {
-		return err
-	}
-
-	newGrub2ConfigFile, err := appendKernelCommandLineArgs(grub2ConfigFile, extraCommandLine)
-	if err != nil {
-		return err
-	}
-
-	// Update grub.cfg file.
-	err = writeGrub2ConfigFile(newGrub2ConfigFile, imageChroot)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// Updates the kernel command-line args with the new SELinux mode.
-//
-// See, installutils.setGrubCfgSELinux()
-func updateSELinuxCommandLine(selinuxMode imagecustomizerapi.SELinuxMode, imageChroot *safechroot.Chroot) error {
-	logger.Log.Infof("Updating SELinux kernel command-line args")
-
-	grub2Config, err := readGrub2ConfigFile(imageChroot)
-	if err != nil {
-		return err
-	}
-
-	newGrub2Config, err := updateSELinuxCommandLineHelper(grub2Config, selinuxMode)
+// Finds all the kernel command-line args that match the provided names, then insert replacement arg(s).
+func updateKernelCommandLineArgs(grub2Config string, argsToRemove []string, newArgs []string) (string, error) {
```

Comment:
- nit: would be good to doc the output
- nit: I wonder if it makes sense to use the same pattern with other function that modify content:
   - `modifySomething(inputGrubConfig string,...) (modifiedGrubConfig string,...)`

Reply:
Fixed.